### PR TITLE
Remove findOrCreate, findOrCreateEach

### PR DESCRIPTION
**Pull request by kevinburkeshyp**
We don't use these functions since they're race-y, and even if we wanted to
implement a find-or-create function, we'd want to a) write it ourselves (e.g.
make it purposefully hard to write) and b) reverse the order, so you attempt
the create first, then find, so we're not vulnerable to races.

### CODE REVIEW

File: lib/waterline/query/dql/update.js [hunk 1]
```diff
@@ -114,21 +114,7 @@ function createBelongsTo(valuesObject, cb) {
...
-    criteria[pkField] = pkValue;
-
-    // If a pkValue if found, do a findOrCreate and look for a record matching the pk.
-    var query;
-    if(pkValue) {
-      query = model.findOrCreate(criteria, valuesObject.values[item]);
-    } else {
-      query = model.create(valuesObject.values[item]);
-    }
+    var query = model.create(valuesObject.values[item]);
```
Comment (dylanpyle) [inline lib/waterline/query/dql/update.js:117]: Just to clarify the change here - was Waterline secretly doing findOrCreates when you made certain other types of queries? Missing a bit of context around this block and the one in `create.js`

**Comment by dylanpyle** [comment]
One question just out of curiosity. The code LGTM though!

File: lib/waterline/query/dql/update.js (hunk 1 above)
Comment (kevinburkeshyp) [inline lib/waterline/query/dql/update.js:117]: As far as I can tell it was/is a way to do like a nested update. so you would update the table and then also as part of your update do like 

``` javascript
.update({email: 'foo@bar.com', relatedTable: {'id': 'dec_123'}) 
```

and then Waterline would find or create the related record for you.

I don't think that we want this feature

Comment (dylanpyle) [inline lib/waterline/query/dql/update.js:117]: Sounds reasonable!

**Pull request closed without merging**